### ISSUE TITLE
Unknown Error if email is not activated

### ISSUE THREAD

**Issue by hargup** [metakgp/metakgp#5]
@Schrodinger1926 reported that he encountered the following error while trying to save an edit, but the error was fixed after he activated his email. @icyflame can you look into this?
![screen shot 2016-06-23 at 02 52 49](https://cloud.githubusercontent.com/assets/2477788/16293955/44d54792-393b-11e6-88ba-ce34b98d0a20.png)

**Comment by icyflame** [comment]
@hargup Is this a bug? Pretty sure that email activation is required for making edits on the wiki. Maybe we can change the error text. That could help.

**Comment by hargup** [comment]
Yes we should provide more helpful error messages. Though I think it shouldn't be necessary to verify the emails of contributors, it can be unnecessary bother at times. Cases of vandalism are quite low, we can revise the policy in future if necessary.

**Comment by hargup** [comment]
@icyflame can you turn off the requirement for email validation I came across another instance today where this proved to be a barrier where I was editing a page without being logged in and the error forced me to log in making me loose the work.

**Comment by icyflame** [comment]
@hargup Really? I can make anonymous edits without doing anything. I don't even have to login to make anon edits through IP.

**Comment by hargup** [comment]
>  I can make anonymous edits without doing anything. I don't even have to login to make anon edits through IP.

This "Unknown Error" occurred while making an anon edit using visual editor, but I cannot reproduce it.

**Comment by amrav** [comment]
Think carefully about whether you want to turn off email validation. That
would mean that user accounts can now be run by spambots more easily.

**Comment by Schrodinger1926** [comment]
@hargup @amrav I would suggest to just show a more informative message when a user tries to make an edit without activating his/her email id. Currently it shows `something went wrong` which gives no clue to user.
I agree with @amrav, turning off email validation doesn't seems to be a good idea.

**Comment by hargup** [comment]
> Think carefully about whether you want to turn off email validation. That would mean that user accounts can now be run by spambots more easily.

We require the users to enter a CAPTCHA, shouldn't it be good enough to fight the bots?

**Comment by icyflame** [comment]
@hargup I don't know if this is a problem anymore. We can show a more informative message, removing email validation isn't a good idea. And since you can't reproduce the problem you had, it's probably a one-off, we can try something if the problem re-appears for someone.

**Comment by icyflame** [comment]
This bug couldn't be reproduced. Was a one-off, most probably. If anyone else faces this again, then we can consider fixing it by tracking the exact events.

Verifying email is required if user wants to have an account on the Wiki. Of course, anon edits can be made with attribution to the IP.

**Comment by hargup** [comment]
Encountered the error with another person, need to investigate this more. See #13
